pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -262,3 +271,40 @@ func (c *DeploymentConfigController) reconcileDeployments(existingDeployments *k
 	}
 	return nil
 }
+
+// cleanupOldDeployments triggers on deployment configuration update and add events in order
+// to delete old replication controller deployments if their quota has been reached
+func (c *DeploymentConfigController) cleanupOldDeployments(existingDeploymentList *kapi.ReplicationControllerList, deploymentConfig *deployapi.DeploymentConfig) error {
+	// we need to coerce the list type into something we can sort
+	existingDeployments := []*kapi.ReplicationController{}
+	for i := 0; i < len(existingDeploymentList.Items); i++ {
+		existingDeployments = append(existingDeployments, &existingDeploymentList.Items[i])
+	}
+
+	// the number of current deployments is one larger than the number of cached existing deployments
+	// as this cleanup is performed only when a new deployment has been created
```

Comment:
A cluster has a bunch of users. Bob has a project with 4 deployments of a dc. The cluster is updated to a new version that has revisionHistoryLimit. Bob updates his spec to try out the new feature (note that he doesn't kick a new deployment) but it doesn't work?